### ISSUE TITLE
[Bug]: Inconsistent return type for `.typeDescriptor()` API for `TypeReferenceTypeSymbol` when it is `IntersectionTypeSymbol`

### ISSUE THREAD

**Issue by lnash94** [ballerina-platform/ballerina-lang#39414]
### Description

When I try to access the type descriptor for the intersection type faces inconsistent behaviour for the below scenarios:
Scenario 01:
```ballerina
public type ReturnValue02 readonly & string;
```
Accesse via:  **TypeSymbol typeSymbol = typeRef.typeDescriptor()**
Return value :  TypeSymbol is  `StringTypeSymbol`

Scenario 02
```ballerina
public type ReturnValue03 readonly & map<json>;

public type ReturnValue04 readonly & json;

public type ReturnValue05 readonly & xml;

public type ReturnValue06 readonly & error;

public type ReturnValue07 readonly & string[];
```
Accesse via:  **TypeSymbol typeSymbol = typeRef.typeDescriptor()**
Return value :  TypeSymbol is  `IntersectionTypeSymbol`

### Steps to Reproduce

Try to access the type descriptor for the below intersection types 
```ballerina
public type ReturnValue02 readonly & string;

public type ReturnValue03 readonly & map<json>;

public type ReturnValue04 readonly & json;

public type ReturnValue05 readonly & xml;

public type ReturnValue06 readonly & error;

public type ReturnValue07 readonly & string[];
```
Accesse via:  **TypeSymbol typeSymbol = typeRef.typeDescriptor()**

### Affected Version(s)

Ballerina 2201.3.2 (Swan Lake Update 3)

### Related area

-> Compilation

### Related issue(s) (optional)

This issue is a blocker for the fix to this issue https://github.com/ballerina-platform/openapi-tools/issues/1151

**Comment by sanjana** [comment]
The reason for this is how the `Intersection` types are handled and returned from the Compiler FE. 

For value types such as `string`, the following condition would be trues and therefore, the `potentialIntersectionType` (in this case types such as `string`) will be returned in the `SymbolResolver`'s `computeIntersectionType()` method.

https://github.com/ballerina-platform/ballerina-lang/blob/8b303cf8b63843bf8f373cb331b8a5e107bcd892/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java#L2375-L2380

**Comment by sanjana** [comment]
> The reason for this is how the `Intersection` types are handled and returned from the Compiler FE.
> 
> For value types such as `string`, the following condition would be trues and therefore, the `potentialIntersectionType` (in this case types such as `string`) will be returned in the `SymbolResolver`'s `computeIntersectionType()` method.
> 
> https://github.com/ballerina-platform/ballerina-lang/blob/8b303cf8b63843bf8f373cb331b8a5e107bcd892/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java#L2375-L2380

The `types.isInherentlyImmutableType(potentialIntersectionType)` will be `true` for `string` type since `types.isValueType(string)` is true.

**Comment by dulajdilshan** [comment]
Inherently Immutable

> Some basic types are inherently immutable: the read-only bit is always on for a value that belongs to an inherently immutable basic type. All simple types are inherently immutable as are functions. Some basic types are selectively immutable: a type is selectively immutable if it is possible to construct both values of the type that have the read-only bit on and values that do not have the read-only bit on. All structured types are selectively immutable as are objects. Finally, some basic types are inherently mutable: the read-only bit is never on for a value belonging to an inherently mutable basic type.

https://ballerina.io/spec/lang/master/#section_5.1.2

**Comment by sanjana** [comment]
The compiler front-end returning the `potentialIntersectionType` for `InherentlyImmutableTypes`, will be addressed in #39585.
Need to validate it in the Semantic API once it is merged to the master.

**Comment by lnash94** [comment]
Remove the blocker label due to we were able to handle the OpenAPI intersection mapping implementation here https://github.com/ballerina-platform/openapi-tools/pull/1189 using the current `.typeDescriptor()` behaviour. Therefore this is not a blocker.
